skybar/src/main/java/org/wtf/skybar/transform/TryCatchMethodVisitor.java

```diff
@@ -0,0 +1,195 @@
+ */
+class TryCatchMethodVisitor extends WorkingLineNumberVisitor {
+
+    private final String className;
+    private final int version;
+    private final int access;
+    private final String desc;
+    private final String sourceFile;
+    private final MethodVisitor mv;
+    private final InsnList instructions;
+    private final LocalVariablesSorter localVariablesSorter;
+
+    private Map<Integer, Integer> lineNumberLocals = new HashMap<>();
+    private Label tryStart;
+    private Label tryEnd;
+    private Label catchHandler;
+    private boolean entered;
+
+
+    public TryCatchMethodVisitor(String className, int version, int access, String desc, String sourceFile, MethodVisitor mv, InsnList instructions, LocalVariablesSorter localVariablesSorter) {
+        super(ASM5, localVariablesSorter);
+        this.className = className;
+        this.version = version;
+        this.access = access;
+        this.desc = desc;
+        this.sourceFile = sourceFile;
+        this.mv = mv;
+        this.instructions = instructions;
+        this.localVariablesSorter = localVariablesSorter;
+    }
+
+    @Override
+    protected void onMethodEnter() {
+
+        getNodesOfType(LineNumberNode.class).forEach(node -> {
+            int lineNumberLocal = localVariablesSorter.newLocal(Type.INT_TYPE);
+            lineNumberLocals.put(node.line, lineNumberLocal);
+            superVisitInsn(Opcodes.ICONST_0);
+            mv.visitVarInsn(Opcodes.ISTORE, lineNumberLocal);
+        });
+        tryStart = new Label();
+        tryEnd = new Label();
+        catchHandler = new Label();
+        mv.visitLabel(tryStart);
+        entered = true;
+    }
+
+
+    @Override
+    protected void onLineNumber(int lineNumber) {
+        SkybarRegistry.registry.registerLine(sourceFile, lineNumber);
+        if(entered) {
+            mv.visitIincInsn(lineNumberLocals.get(lineNumber), 1);
+        } else {
+            if(useInvokeDynamic()) {
+                MethodType mt = MethodType.methodType(CallSite.class, MethodHandles.Lookup.class, String.class, MethodType.class, String.class, int.class);
+
+                Handle bootstrap = new Handle(Opcodes.H_INVOKESTATIC, Type.getInternalName(SkybarRegistry.class), "bootstrap",
+                        mt.toMethodDescriptorString());
+                mv.visitInvokeDynamicInsn("visitLine", "()V", bootstrap, sourceFile, lineNumber);
+            } else {
+                mv.visitFieldInsn(GETSTATIC, "org/wtf/skybar/registry/SkybarRegistry", "registry", "Lorg/wtf/skybar/registry/SkybarRegistry;");
+                mv.visitLdcInsn(sourceFile);
+                mv.visitLdcInsn(lineNumber);
+                mv.visitMethodInsn(INVOKEVIRTUAL, "org/wtf/skybar/registry/SkybarRegistry", "getAdderForLine", "(Ljava/lang/String;I)Ljava/util/concurrent/atomic/LongAdder;", false);
+                mv.visitInsn(LCONST_1);
+                mv.visitMethodInsn(INVOKEVIRTUAL, "java/util/concurrent/atomic/LongAdder", "add", "(J)V", false);
```

Comment:
Nope; it would have been more useful to point out here that I originally only noticed the sharing for the if {} clause. :)